### PR TITLE
 🐛Fix `RuntimeError` raised when `HTTPException` has a status code with no content

### PR DESCRIPTION
**Pull request by iudeen**
Fixes/related to https://github.com/tiangolo/fastapi/issues/4949#issuecomment-1241143806

Please advice if this approach is correct one.

Suggestions welcome!

### CODE REVIEW

File: tests/test_starlette_exception.py [hunk 1]
```diff
@@ -18,6 +18,18 @@ async def read_item(item_id: str):
     return {"item": items[item_id]}
 
 
+@app.get("/http-no-body-statuscode-exception")
+async def no_body_status_code_exception():
+    raise HTTPException(
+        status_code=204,
+    )
```
Comment (Kludex) [inline tests/test_starlette_exception.py:25]:
```suggestion
    raise HTTPException(status_code=204)
```

**Review by Kludex — APPROVED** [review]
Yep, same idea as what we have in Starlette: https://github.com/encode/starlette/blob/bc61505faef15b673bf4bf65db4927daa354d6b8/starlette/middleware/exceptions.py#L99-L104

**Comment by iudeen** [comment]
> Yep, same idea as what we have in Starlette: https://github.com/encode/starlette/blob/bc61505faef15b673bf4bf65db4927daa354d6b8/starlette/middleware/exceptions.py#L99-L104

The way I was able to quickly find the bug was adding debug points to ExceptionMiddleware while running the app using FastAPI and Starlette. 

And that code was from Starlette! (Sorry, should have mentioned)

**Review by BilalAlpaslan — APPROVED** [review]

**Review by JarroVGIT — APPROVED** [review]
LGTM

**Comment by tiangolo** [comment]
Awesome, thanks for the fix @iudeen! :cake: :rocket: 

And thanks for the reviews and comments @Kludex, @JarroVGIT, @BilalAlpaslan :bow: :coffee:

**Pull request merged**